src/app/concern/file-uploader/file-uploader.component.ts

```diff
@@ -38,31 +46,46 @@ export class FileUploaderComponent implements OnInit {
     this.dropArea.nativeElement.classList.remove("highlight");
   }
 
-  public handleDrop(e: any): void {
-    // const dt = e.dataTransfer;
-    // this.uploadedFiles = dt.files;
-    this.upload();
+  public handleDrop($event: DragEvent): void {
+    const itemList: DataTransferItem[] = Array.from($event.dataTransfer.items);
+    const filesAllowed: string[] = this.acceptedFileTypes.split(",");
+    const droppedFiles: File[] = [];
+    const invalidFiles: string[] = [];
+
+    if (itemList.length) {
+      // TODO remove check once multiple file upload is supported by BE
+      if (itemList.length > 1) {
+        this.snackBarService.openSnackBar(`Please select one file at a time`);
+        return;
+      }
+
+      itemList.forEach((i: DataTransferItem) => {
+        if (filesAllowed.includes(i.type)) {
+          droppedFiles.push(i.getAsFile());
+        } else {
+          invalidFiles.push(i.getAsFile().name);
+        }
+      });
+
+      if (invalidFiles.length) {
```

Comment:
We only have one file at a time and would push the same to droppedFiles / InvalidFiles. Why do need to push them in list and check length? Can't we return from above if.

Reply:
very true, BE will be enabling multiple file upload feature shortly so built the FE with a list in mind : )

Reply:
I would prefer to add the code later as future is very uncertain ;)